### ISSUE TITLE
Can't run any of the game_2048 demo scripts on Ubuntu 24.04, Python 3.12.3

### ISSUE THREAD

**Issue by Jaccxc** [opendilab/LightZero#303]
Hi, thank you all for the great work on this repository! I'm very new to reinforcement learning, so please correct me if I’ve made any mistakes!

## Describe the bug
Can't run any of the game_2048 demo scripts(muzero_2048_config.py and stochastic_muzero_2048_config.py). 

I have successfully run the Cartpole but not this game_2048 demo.

After the first episode ended, and right after learner save ckpt, the error of indicating mismatch of batch_rewords shape shows up. Then the whole error is followed by the connection between process failed. And that's about the whole error.

Complete log around the error:
```sh
[12-09 05:09:18] INFO     [EVALUATOR]env 1 finish episode, final reward: 108.0, current episode: 1     muzero_evaluator.py:353
[12-09 05:09:18] INFO     [EVALUATOR]env 0 finish episode, final reward: 156.0, current episode: 2     muzero_evaluator.py:353
[12-09 05:09:19] INFO     [EVALUATOR]env 2 finish episode, final reward: 248.0, current episode: 3     muzero_evaluator.py:353
[12-09 05:09:19] INFO                                                                                  muzero_evaluator.py:429
                          +-------+------------+---------------------+---------------+---------------+                        
                          | Name  | train_iter | ckpt_name           | episode_count | envstep_count |                        
                          +-------+------------+---------------------+---------------+---------------+                        
                          | Value | 0.000000   | iteration_0.pth.tar | 3.000000      | 127.000000    |                        
                          +-------+------------+---------------------+---------------+---------------+                        
                          +-------+-------------------------+---------------+---------------------+---                        
                          -------------------+                                                                                
                          | Name  | avg_envstep_per_episode | evaluate_time | avg_envstep_per_sec |                           
                          avg_time_per_episode |                                                                              
                          +-------+-------------------------+---------------+---------------------+---                        
                          -------------------+                                                                                
                          | Value | 42.333333               | 4.324197      | 29.369613           |                           
                          0.693770             |                                                                              
                          +-------+-------------------------+---------------+---------------------+---                        
                          -------------------+                                                                                
                          +-------+-------------+------------+------------+------------+                                      
                          | Name  | reward_mean | reward_std | reward_max | reward_min |                                      
                          +-------+-------------+------------+------------+------------+                                      
                          | Value | 170.666667  | 58.088056  | 248.000000 | 108.000000 |                                      
                          +-------+-------------+------------+------------+------------+                                      
                          +-------+-----------------------+--------------------------+                                        
                          | Name  | eval_episode_return   | eval_episode_return_mean |                                        
                          +-------+-----------------------+--------------------------+                                        
                          | Value | [156.0, 108.0, 248.0] | 170.666667               |                                        
                          +-------+-----------------------+--------------------------+                                        
                                                                                                                              
[12-09 05:09:19] INFO     [RANK0]: learner save ckpt in                                                    base_learner.py:338
                          ./data_muzero/game_2048_npct-2_muzero_ns100_upc200_rer0.0_bs512_sslw2_seed0_2412                    
                          09_050914/ckpt/ckpt_best.pth.tar                                                                    
Traceback (most recent call last):
  File "/home/jacc/LightZero/zoo/game_2048/config/muzero_2048_config.py", line 92, in <module>
    train_muzero([main_config, create_config], seed=0, model_path=main_config.policy.model_path, max_env_step=max_env_step)
  File "/home/jacc/LightZero/lzero/entry/train_muzero.py", line 205, in train_muzero
    train_data = replay_buffer.sample(batch_size, policy)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jacc/LightZero/lzero/mcts/buffer/game_buffer_muzero.py", line 141, in sample
    batch_rewards, batch_target_values = self._compute_target_reward_value(
                                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jacc/LightZero/lzero/mcts/buffer/game_buffer_muzero.py", line 525, in _compute_target_reward_value
    batch_rewards = np.asarray(batch_rewards)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 2 dimensions. The detected shape was (512, 6) + inhomogeneous part.
Exception ignored in: <function MuZeroCollector.__del__ at 0x7dd601c0f6a0>
Traceback (most recent call last):
  File "/home/jacc/LightZero/lzero/worker/muzero_collector.py", line 196, in __del__
    self.close()
  File "/home/jacc/LightZero/lzero/worker/muzero_collector.py", line 185, in close
    self._env.close()
  File "/home/jacc/.venv/jacc_light_zero/lib/python3.12/site-packages/ding/envs/env_manager/subprocess_env_manager.py", line 635, in close
    p.send(['close', None, None])
  File "/usr/lib/python3.12/multiprocessing/connection.py", line 206, in send
    self._send_bytes(_ForkingPickler.dumps(obj))
  File "/usr/lib/python3.12/multiprocessing/connection.py", line 427, in _send_bytes
    self._send(header + buf)
  File "/usr/lib/python3.12/multiprocessing/connection.py", line 384, in _send
    n = write(self._handle, buf)
        ^^^^^^^^^^^^^^^^^^^^^^^^
BrokenPipeError: [Errno 32] Broken pipe
Exception ignored in: <function MuZeroEvaluator.__del__ at 0x7dd601c1c900>
Traceback (most recent call last):
  File "/home/jacc/LightZero/lzero/worker/muzero_evaluator.py", line 174, in __del__
    self.close()
  File "/home/jacc/LightZero/lzero/worker/muzero_evaluator.py", line 163, in close
    self._env.close()
  File "/home/jacc/.venv/jacc_light_zero/lib/python3.12/site-packages/ding/envs/env_manager/subprocess_env_manager.py", line 635, in close
    p.send(['close', None, None])
  File "/usr/lib/python3.12/multiprocessing/connection.py", line 206, in send
    self._send_bytes(_ForkingPickler.dumps(obj))
  File "/usr/lib/python3.12/multiprocessing/connection.py", line 427, in _send_bytes
    self._send(header + buf)
  File "/usr/lib/python3.12/multiprocessing/connection.py", line 384, in _send
    n = write(self._handle, buf)
        ^^^^^^^^^^^^^^^^^^^^^^^^
BrokenPipeError: [Errno 32] Broken pipe
 
```

## To Reproduce
I'm using a freshly installed Ubuntu 24.04 system with Python 3.12.3 out of the box. Then I used venv to create the environment. Other than that, I followed the installation steps and also manually installed "2to3" package using apt. Finally run the game_2048 config, and that's about it!  
```properties
git clone https://github.com/opendilab/LightZero.git
cd LightZero
pip3 install -e .
python3 -u zoo/game_2048/config/muzero_2048_config.py
```

## Expected behavior
I had successfully run the Cartpole demo. Therefore this should also be working.

## Desktop
 - OS: Ubuntu 24.04 freshly installed
 - Python version:  Python 3.12.3

Thank you all for the work again!

**Comment by Jaccxc** [comment]
Didn't notice that @walkacross just posted about this bug.

**Comment by puyuan1996** [comment]
Thank you very much for your feedback and suggestions! You can refer to the fixes and updates discussed in [GitHub Issue #302](https://github.com/opendilab/LightZero/issues/302) for details on resolving the mentioned bugs. If you encounter further issues or have additional suggestions, please feel free to let us know. We greatly appreciate your contributions!
